lib/db/schemaupdater.go

```diff
@@ -835,6 +837,59 @@ func (db *schemaUpdater) dropIndexIDsMigration(_ int) error {
 	return db.dropIndexIDs()
 }
 
+func (db *schemaUpdater) fixRecvEncFileSize(_ int) error {
+	meta := newMetadataTracker(db.keyer, db.evLogger)
+	t, err := db.newReadWriteTransaction()
+	if err != nil {
+		return err
+	}
+	defer t.close()
+
+	var key []byte
+	for _, folderStr := range db.ListFolders() {
+		folder := []byte(folderStr)
+		var innerErr error
+		encParent := ""
+		err := t.withHave(folder, protocol.LocalDeviceID[:], nil, false, func(fi protocol.FileIntf) bool {
+			f := fi.(protocol.FileInfo)
+			switch f.Type {
+			case protocol.FileInfoTypeDirectory:
+				if protocol.IsEncryptedParent(fs.PathComponents(f.Name)) {
+					encParent = f.Name
+				} else {
+					encParent = ""
+				}
+				return true
+			case protocol.FileInfoTypeFile:
+			default:
+				return true
+			}
+			if !strings.HasPrefix(f.Name, encParent) {
```

Comment:
This one is cool too because it matches all files before we see an encrypted parent, because `encParent` starts out empty... I was wondering why all my files were suddenly local additions :D